go/vt/schemadiff/schema.go

```diff
@@ -198,12 +212,57 @@ func (s *Schema) normalize() error {
 		return true
 	}
 
+	// We now iterate all tables. We iterate "dependency levels":
+	// - first we want all tables that don't have foreign keys or which only reference themselves
+	// - then we only want tables that reference 1st level tables. these are 2nd level tables
+	// - etc.
+	// we stop when we have been unable to find a table in an iteration.
+	fkParents := map[string]bool{}
+	iterationLevel := 0
+	for ; ; iterationLevel++ {
```

Comment:
I'd probably write this slight different like this:

```go
for {
     iterationLevel++ 
     ...
}
```

Reads simpler to me, but no strong opinion really on this. 

The other question though is, would it be possible to craft a schema manually that triggers an infinite loop here? Should we have a safeguard for a max level we support?

Reply:
> would it be possible to craft a schema manually that triggers an infinite loop

The algorithm addresses infinite loops. It leaves out any infinitely dependent views. There's a unit test for that: https://github.com/vitessio/vitess/blob/ec7e1a875224507ffd026b180f793205126cb336/go/vt/schemadiff/schema_test.go#L126-L135

Reply:
just clarifying, that the rewrite would be:

```go
for {
     ...
     iterationLevel++ 
}
```

Reply:
fixed